### DISCUSSION TITLE
Changing the name, 'Cells'

### DISCUSSION THREAD

**Discussion by fumitoh** [fumitoh/modelx#148] [Ideas]
I've been tempted to change the name 'Cells' for quite some time. The original name was chosen by analogy with spreadsheet cells, but it has since become a misnomer. After exploring and rejecting several alternative names—such as 'Method,' 'Calc,' 'SmartFunc,' and 'Memoizer'—the most promising candidate at the moment is 'Functor,' a term historically used in C++ to mean a function object. Consequently, some API functions and methods, such as `defcells`, `new_cells`, and `clear_cells`, would be renamed. The old names would also remain available for quite a while.

**Comment by AB-Athene** [comment]
I like 'Cells', 'Functor' sounds scary

**Reply by fumitoh** [reply]
What do you mean by scary? Are you concerned about possible incompatibility issues related to the name change?

**Reply by alexeybaran** [reply]
When moving actuaries from Excel it is better to make Python and `modelx` sound easy. `Functor` sounds very scientific.
`Cells` is a good term as it reflects that it is part of the grid (of dependencies) and is clearly differentiated from "function".
Yes, there will be some compatibility issues

**Comment by bakerwy** [comment]
Functor sounds cool. But is there reason using functor instead of function? Because function sounds more intuitive.

**Reply by fumitoh** [reply]
Function could be a candidate. But  It would be not clear in documentation whether "function" means a modelx function or Python function.
